### PR TITLE
Rename SpanContext to SpanReference

### PR DESCRIPTION
**Pull request by drexler**
Reference issue: #267

### CODE REVIEW

**Comment by drexler** [comment]
Probably a delayed response but I've already signed the EasyCLA Individual Contributor Agreement

File: src/sdk/propagation/trace_context.rs [hunk 1]
```diff
@@ -102,42 +102,42 @@ impl TraceContextPropagator {
...
-    /// OR if the retrieved SpanContext is invalid then an empty `SpanContext`
+    /// Retrieves encoded `SpanReference`s using the `Extractor`. It decodes
+    /// the `SpanReference` and returns it. If no `SpanReference` was retrieved
+    /// OR if the retrieved SpanReference is invalid then an empty `SpanReference`
     /// is returned.
     fn extract_with_context(&self, cx: &Context, extractor: &dyn Extractor) -> Context {
-        self.extract_span_context(extractor)
-            .map(|sc| cx.with_remote_span_context(sc))
+        self.extract_span_reference(extractor)
+            .map(|sc| cx.with_remote_span_reference(sc))
```
Comment (jtescher) [inline src/sdk/propagation/trace_context.rs:140]: Small nit: this should be `sr` now instead of `sc`

**Review by jtescher — APPROVED** [review]
Looks good thanks @drexler, if you push a commit I think it will bump the CLA bot 🙄

File: src/sdk/propagation/trace_context.rs (hunk 1 above)
Comment (drexler) [reply src/sdk/propagation/trace_context.rs:140]: good catch!

**Comment by drexler** [comment]
@jtescher Thanks for the review 👍

**Pull request merged**